backend/tracim_backend/command/cleanup.py

```diff
@@ -0,0 +1,222 @@
+        )
+        parser.add_argument(
+            "-f",
+            "--force",
+            help="force user deletion",
+            dest="force",
+            default=False,
+            action="store_true",
+        )
+        parser.add_argument(
+            "-l", "--login", nargs="+", help="User logins (email)", dest="logins", required=True
+        )
+        return parser
+
+    def take_app_action(self, parsed_args: argparse.Namespace, app_context: AppEnvironment) -> None:
+        self._session = app_context["request"].dbsession
+        self._app_config = app_context["registry"].settings["CFG"]
+
+        if parsed_args.test_mode:
+            print("(!) Running in Test mode, not change will be applied.")
+        if parsed_args.force:
+            print("/!\\ Running in force mode, database created may be broken /!\\.")
+        elif parsed_args.best_effort:
+            print(
+                "(!) Running in best effort mode, will anonymise account instead of deleting them if needed."
+            )
+
+        deleted_user_ids = set()
+        deleted_workspace_ids = set()
+        with unprotected_content_revision(self._session) as session:
+            uapi = UserApi(
+                config=self._app_config,
+                session=session,
+                current_user=None,
+                show_deleted=True,
+                show_deactivated=True,
+            )
+            user_list = []  # type: typing.List[User]
+            for login in parsed_args.logins:
+                try:
+                    user = uapi.get_one_by_email(login)
+                    user_list.append(user)
+                except UserDoesNotExist:
+                    print("ERROR: user with email {} does not exist".format(login))
+                    exit(2)
+            print("~~~~~~~~~~")
+            for user in user_list:
+                cleanup_lib = CleanupLib(session, self._app_config, test_mode=parsed_args.test_mode)
+                deleted_user_ids_result = self._delete_user_database_info(
+                    user,
+                    force=parsed_args.force,
+                    best_effort=parsed_args.best_effort,
+                    cleanup_lib=cleanup_lib,
+                )
+                deleted_user_ids.add(deleted_user_ids_result.user_id)
+                deleted_workspace_ids.update(deleted_workspace_ids)
+                print("~~~~~~~~~~")
+
+            # INFO - G.M - 2019-12-13 - cleanup agenda at end of process
+            if deleted_workspace_ids:
+                deleted_workspace_ids_str = [
+                    '"{}"'.format(workspace_id) for workspace_id in deleted_workspace_ids
+                ]
+                print("Delete agenda of workspaces {}".format(", ".join(deleted_workspace_ids_str)))
+                for workspace_id in deleted_workspace_ids:
+                    try:
+                        cleanup_lib.delete_workspace_agenda(workspace_id)
+                    except FileNotFoundError:
```

Comment:
Not really nice but not sure how we should handle "agenda does not exist case. It's weird to block workspace deletion if agenda doesn't exist. By the way, workspace can have agenda feature disabled by when we delete agenda here, we do not have anymore the information. Retrieving it is possible but complicate thing.

Reply:
No problem with process continuation when workspace already deleted. But i suggest using a more precise exception class at this level, eg. `AlreadyDeletedAgendaError`.